File: 프로그래머스/lv3/42579. 베스트앨범/베스트앨범.py
def solution(genres, plays):
    music = []
    g = {}
    for i in range(len(plays)):
        music.append([genres[i],plays[i],i])
        if not genres[i] in g:
            g[genres[i]] = plays[i]
        else:
            g[genres[i]] += plays[i]
    li = []
    for a,b in g.items():
        li.append([a,b])
    li.sort(key = lambda x : (x[1],x[0]), reverse = True)
    music.sort(key = lambda x: (x[1]), reverse = True)
    arr = []
    for i in range(len(li)):
        cnt = 0
        for j in range(len(music)):
            if li[i][0] == music[j][0]:
                arr.append(music[j][2])
                cnt+=1
            if cnt==2:
                break
    return arr

File: 프로그래머스/3/42579. 베스트앨범/베스트앨범.js
function solution(genres, plays) {
    const songs = []
    const map = new Map()
    for(let i=0;i<genres.length;i++){
        songs.push([genres[i],plays[i],i])
    }
    for(let song of songs){
        if(map.has(song[0])) map.set(song[0],map.get(song[0])+song[1])
        else map.set(song[0],song[1])
    }
    const genresWithPlay = [...map].sort((a,b) => (b[1]-a[1]))
    const ans = []
    songs.sort((a,b)=>(b[1]-a[1]))
    console.log(songs)
    console.log(genresWithPlay)
    for(let i=0;i<genresWithPlay.length;i++){
        for(let j=0;j<2;j++){
            for(let k=0;k<songs.length;k++){
                if(genresWithPlay[i][0] === songs[k][0] && !ans.includes(songs[k][2])) {
                    ans.push(songs[k][2])
                    break
                }
            }
        }
    }
    return ans
}